15210-Fixed product tier pricing pagination issue in admin

saravananvelu opened the pull request:
Resolved product tier price pagination loader issue in admin

Description:

Navigate Admin => Catalog => Products => Create New / Edit Product.

1. Go to Advanced Pricing link under Price text field.
2. Add tier price more than a page and do save.
3. Jump to other pages by directly editing the pagination text box.


### Fixed Issues (if relevant)

1. magento/magento2#15210: Advanced pricing pagination issue

saravananvelu committed:
15210-Fixed product tier pricing pagination issue in admin

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=15360) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=15360) before we can accept your contribution.<br/>

gelanivishal committed:
Fixed coding related issue

sidolov commented:
Hi @saravananvelu  , please, sign CLA, otherwise, we can't process your pull request.

gelanivishal committed:
Removed Illegal trailing whitespace.

saravananvelu committed:
Removed newly added comments

saravananvelu committed:
Inherited the parent method to fix pagination issue

magento-engcom-team commented:
@saravananvelu thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

VladimirZaets commented on app/code/Magento/Catalog/view/adminhtml/web/js/components/dynamic-rows-tier-price.js:
```diff
@@ -28,6 +28,33 @@ define([
+        changePage: function (page) {
+            this.clear();
+
+            if (page === 1 && !this.recordData().length) {
```
Hi @saravananvelu. Thanks for the collaboration.
This issue reproduced in all places where dynamic-rows component is used, so will be great if you fix this issue in the more general way, in the parent class. 

Also, you can call the method from the parent class (`this._super()`) instead of copy-paste it

saravananvelu replied:
I have updated the code in parent class and removed the same from child class

saravananvelu committed:
Fixed product tier pricing pagination issue in admin

VladimirZaets committed:
15210-Fixed product tier pricing pagination

- Fix prevPage and nextPage method. Calling clear method was added to changePage method.

VladimirZaets commented:
Hi @saravananvelu, thanks! 
Also, I removed calling the clear method inside previous and next page method. Currently, it will be redundant.

magento-engcom-team commented:
Hi @saravananvelu. Thank you for your contribution. 
Changes from your Pull Request will be available with the upcoming 2.3.0 release.
